pkg/cmd/pkg/push.go

```diff
@@ -0,0 +1,159 @@
+package pkg
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/go-buildkite/v3/buildkite"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type pushPackageConfig struct {
+	RegistrySlug  string
+	FilePath      string
+	StdinFileName string
+}
+
+const stdinFileNameFlag = "stdin-file-name"
+
+func NewCmdPackagePush(f *factory.Factory) *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "push registry-name {path/to/file | --stdin-file-name filename -}",
+		Short: "Push a new package to Buildkite Packages",
+		Long: heredoc.Doc(`
+			Push a new package to Buildkite Packages. The package can be passed as a path to a file in the second positional argument,
+			or via stdin. If passed via stdin, the filename must be provided with the --stdin-file-name flag, as Buildkite
+			Packages requires a filename for the package.`),
+		Example: heredoc.Doc(`
+			$ bk package push my-registry my-package.tar.gz
+			$ cat my-package.tar.gz | bk package push my-registry --stdin-file-name my-package.tar.gz - # Pass package via stdin, note hyphen as the argument
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, err := loadAndValidateConfig(cmd.Flags(), args)
+			if err != nil {
+				return fmt.Errorf("failed to validate flags and args: %w", err)
+			}
+
+			packageName := ""
+			var from io.Reader
+			switch {
+			case cfg.FilePath != "":
+				packageName = cfg.FilePath
+
+				file, err := os.Open(cfg.FilePath)
+				if err != nil {
+					return fmt.Errorf("couldn't open file %s: %w", cfg.FilePath, err)
+				}
+
+				defer file.Close()
+
+				from = file
+			case cfg.StdinFileName != "":
+				packageName = cfg.StdinFileName
+				from = cmd.InOrStdin()
+			}
+
+			pkg, _, err := f.RestAPIClient.PackagesService.Create(f.Config.OrganizationSlug(), cfg.RegistrySlug, buildkite.CreatePackageInput{
+				Filename: packageName,
+				Package:  from,
+			})
```

Comment:
Should we add a check to ensure this isn't a nil pointer?

Reply:
yeah, i was wondering about that earlier... theoretically `loadAndValidateConfig` should ensure that one or the other of the two sources is selected, but it can't hurt to double check.

given that neither stdin nor a file being available is likely an error in programming, i add a 
```
default:
  panic("oh no!")
```
to the switch block above.

Reply:
Is this upload where the bulk of the time will be spent? Ie if its uploading a few MB or many more, it could take a while. We have a spinner package we use to give the user some feedback at least. I made a patch you can apply if you like: https://gist.github.com/jradtilbrook/a3af75bab23d26edb6f2bb2291bf17d1

I'm assuming simply opening the file won't take long and not worth including in the spinner